testdata/client/connecth1unaryerror

```diff
@@ -0,0 +1,59 @@
+{
+  "testName": "Unary with an Error",
+  "httpVersion": "HTTP_VERSION_1",
+  "protocol": "PROTOCOL_CONNECT",
+  "codec": "CODEC_JSON",
+  "host": "127.0.0.1",
+  "port": 51356,
+  "service": "connectrpc.conformance.v1alpha1.ConformanceService",
+  "method": "Unary",
+  "streamType": "STREAM_TYPE_UNARY",
+  "requestHeaders": [
+    {
+      "name": "X-Conformance-Test",
+      "value": [
+        "Value1",
+        "Value2"
+      ]
+    }
+  ],
+  "requestMessages": [
+    {
+      "@type": "type.googleapis.com/connectrpc.conformance.v1alpha1.UnaryRequest",
+      "responseDefinition": {
+        "responseHeaders": [
+          {
+            "name": "x-custom-header",
+            "value": [
+              "foo",
+              "bar",
+              "baz"
+            ]
+          }
+        ],
+        "error": {
+          "code": 13,
+          "message": "unary failed",
+          "details": [
+            {
+              "@type": "type.googleapis.com/connectrpc.conformance.v1alpha1.Header",
```

Comment:
Used a `Header` type here just to test. I wanted to use a more real-world example such as `google.rpc.RetryInfo`, but I kept getting 'unable to resolve' errors and punted on it.

Reply:
Yep. Any real-world examples must refer to messages that your program actually knows about. To avoid those kinds of dependencies (and so that other client and server implementations that will be under test don't need to import/link in lots of other generated proto files/packages), I think it's best if we stick to messages that are defined in these conformance protos.

Reply:
Gotcha. I actually defined a `RetryInfo` in our protos too and got that same message, but I assume I was just doing something wrong.

Reply:
If you defined it in a different package, that would be the issue: the Go runtime is only to do JSON decoding of `Any` messages for messages defined in packages that are actually _imported_ by the program. So if you added new messages in a different folder, but then never updated any of this code to _import_ that package, the types will not be available.

FWIW, this is only an issue when decoding JSON. For the protobuf binary format, `Any` messages can be used and de-serialized without knowing the schema of the message therein. So we should actually test this in a test case, that we the server can reply correctly even if an error details contains an unknown message. To do that, the test runner itself will need to know about some protos that the clients/servers under test (including the reference client and server) do _not_ know about, so that it can actually decode them (from YAML configs).

But I think we can worry about that later. We could even have the test runner _dynamically_ load (or even generate) some protos, so it's _guaranteed_ that the client or server under test _can't_ know about them, and then verify that it can return the errors correctly anyhow.

Anyhow, I say all of this because the actual gRPC/gRPC-Web/Connect format for errors does not require the server or client knowing the schema, since they use the binary format.